dict_scripts/sss_compress.py

```diff
@@ -0,0 +1,80 @@
+
+
+def g(s, d):
+    code_page = """¡¢£¤¥¦©¬®µ½¿€ÆÇÐÑ×ØŒÞßæçðıȷñ÷øœþ !"#$%&'()*+,-./0123456789:;<=>?@ABCDEFGHIJKLMNOPQRSTUVWXYZ[\]^_`abcdefghijklmnopqrstuvwxyz{|}~¶"""
+    code_page += """°¹²³⁴⁵⁶⁷⁸⁹⁺⁻⁼⁽⁾ƁƇƊƑƓƘⱮƝƤƬƲȤɓƈɗƒɠɦƙɱɲƥʠɼʂƭʋȥẠḄḌẸḤỊḲḶṂṆỌṚṢṬỤṾẈỴẒȦḂĊḊĖḞĠḢİĿṀṄȮṖṘṠṪẆẊẎŻạḅḍẹḥịḳḷṃṇọṛṣṭ§Äẉỵẓȧḃċḋėḟġḣŀṁṅȯṗṙṡṫẇẋẏż«»‘’“”"""
+
+    # constant. (exclusive)
+    end_len = 2 + max(
+        len(w) for w in d["long"]
+    )  # no need to consider dictionary.short because obviously, it's shorter
+
+    # Increase efficiency without implementing a trie (or sth similar)
+    d["short"] = {v: i for i, v in enumerate(d["short"])}
+    d["long"] = {v: i for i, v in enumerate(d["long"])}
+
+    def character(z, c):
+        if c in "\n\x7f¶":
+            o = 95
+        elif " " <= c <= "~":
+            o = ord(c) - 32
+        else:
+            raise ValueError(c + " is neither printable ASCII nor a linefeed.")
+        z = 96 * z + o
+        z = 3 * z + 0
+        return z
+
+    def dictionary(z, w, nonempty):
+        ts = nonempty
+        if w[:1] == " ":
+            w = w[1:]
+            ts = not ts
+        dct = d["short"] if len(w) < 6 else d["long"]
+        W, sc = (w, 0) if w in dct else (w[:1].swapcase() + w[1:], 1)
+        if W not in dct:
+            raise ValueError(w + " isn't in the dictionary.")
+        f = ts or sc
+        j = (2 if sc else 1) if ts else 0
+        i = dct[W]
+
+        z = len(dct) * z + i
+        z = 2 * z + int(len(w) < 6)
+        if f:
+            z = 3 * z + j
+            z = 3 * z + 2
+        else:
+            z = 3 * z + 1
+        return z
+
+    def go(z):
+        compressed = []
+        while z:
+            z, c = divmod(
+                z - 1, 252
+            )  # Taken inspiration from Jelly's bijective base conversion algorithm
+            compressed.append(code_page[c])
+        return "“{0}»".format("".join(compressed[::-1]))
+
+    def optimal(str):
+        dp = [0] * -~len(str)
+        for i in range(len(str) - 1, -1, -1):
+            dp[i] = character(dp[i + 1], str[i])
+            for j in range(1, min(len(dp) - i, end_len)):
+                try:
+                    dp[i] = min(
+                        dp[i],
+                        dictionary(dp[i + j], str[i : i + j], i != 0),
+                    )
+                except ValueError:
+                    pass
+
+        return min(
+            [
+                "“{0}”".format(str),
```

Comment:
```suggestion
        return '"{0}”'.format("".join(compressed[::-1]))

    def optimal(str):
        dp = [0] * -~len(str)
        for i in range(len(str) - 1, -1, -1):
            dp[i] = character(dp[i + 1], str[i])
            for j in range(1, min(len(dp) - i, end_len)):
                try:
                    dp[i] = min(
                        dp[i],
                        dictionary(dp[i + j], str[i : i + j], i != 0),
                    )
                except ValueError:
                    pass

        return min(
            [
                '"{0}"'.format(str),
```

wrong apostrophes?

Reply:
No, this is Jelly's string delimiters.

Reply:
This is only to get length under compression. The strings produced are irrelevant.